kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/api/StartupStrategy.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.services.api;
+
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.kie.server.api.model.KieContainerResource;
+import org.kie.server.services.impl.ContainerManager;
+import org.kie.server.services.impl.KieServerImpl;
+import org.kie.server.services.impl.storage.KieServerState;
+
+public interface StartupStrategy {
+
+    void startup(KieServerImpl kieServer, ContainerManager containerManager, KieServerState currentState, AtomicBoolean kieServerActive);
+    
+    default Set<KieContainerResource> prepareContainers(Set<KieContainerResource> containers) {
```

Comment:
What is the purpose of this method?

Reply:
this is to allow any preparation like sorting (https://issues.jboss.org/browse/JBPM-7051) of containers before they are deployed or whatever else you would like to do - filter out some containers etc.

Reply:
Ok, thanks